### PR TITLE
improve and add missing docs

### PR DESCRIPTION
**Pull request by mindplay-dk**
Added the missing `ondestroy` event, updated the docs for the `onremove` event, clarified use of events with use-case examples, moved the `patch` function documentation up with the rest of the API description in the "Usage" section, documented the `key` attribute and when/how to use it, etc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,43 +81,69 @@ function AwesomeTitle(text) {
...
+)
+```
+
+The `patch` function returns the patched child element.
+
+## Lifecycle Events
+
+Picodom supports element-level life-cycle events and keyed updates, and reserves the following JSX attributes:
+
+#### `key`
```
Comment (JorgeBucaran) [inline README.md]: @mindplay-dk `key` is not a lifecycle event though. Can you put this somewhere else?

Comment (mindplay-dk) [reply README.md]: But it's part of the life-cycle control, right?

It has no use other than to control the life-cycle of the element, I think?

So maybe the title of that section should be "Life-cycle Control", not "Life-cycle Events"?

Or maybe just "Reserved Attributes"?

Comment (JorgeBucaran) [reply README.md]: Yeah, reserved attributes sounds 👍

**Comment by mindplay-dk** [comment]
@JorgeBucaran I'd like to document support for `class` and special handling of `style` too, so that all of attribute handling is fully documented - do you think I should do that under the same heading as the other attributes, or separately?

Maybe change the heading to something more general like "Supported Attributes" or "Special Attributes"?

**Comment by JorgeBucaran** [comment]
@mindplay-dk Sounds good. "Supported Attributes" is fine 👍

**Comment by mindplay-dk** [comment]
Good to go?

**Comment by JorgeBucaran** [comment]
Good good! 🎉

**Pull request merged**